From f171a4c757ac332edde4246ac178f9efb0fdafe7 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Wed, 11 Dec 2024 20:41:30 +0100
Subject: [PATCH 24/24] Revert "x86/cpu: Enable SD_ASYM_PACKING for PKG domain
 on systems with AMD preferred cores"

This reverts commit 912c9fd7a03b0009cc98eac95fb5ffc91c97f365.
---
 arch/x86/kernel/smpboot.c | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/arch/x86/kernel/smpboot.c b/arch/x86/kernel/smpboot.c
index 419e7ae09..b5a8f0891 100644
--- a/arch/x86/kernel/smpboot.c
+++ b/arch/x86/kernel/smpboot.c
@@ -62,8 +62,6 @@
 #include <linux/mc146818rtc.h>
 #include <linux/acpi.h>
 
-#include <acpi/cppc_acpi.h>
-
 #include <asm/acpi.h>
 #include <asm/cacheinfo.h>
 #include <asm/desc.h>
@@ -499,19 +497,10 @@ static int x86_cluster_flags(void)
 
 static int x86_die_flags(void)
 {
-	if (cpu_feature_enabled(X86_FEATURE_HYBRID_CPU))
+	if (cpu_feature_enabled(X86_FEATURE_HYBRID_CPU) ||
+	    cpu_feature_enabled(X86_FEATURE_AMD_HETEROGENEOUS_CORES))
 		return x86_sched_itmt_flags();
 
-	switch (boot_cpu_data.x86_vendor) {
-	case X86_VENDOR_AMD:
-	case X86_VENDOR_HYGON:
-		bool prefcore = false;
-
-		amd_detect_prefcore(&prefcore);
-		if (prefcore || cpu_feature_enabled(X86_FEATURE_AMD_HETEROGENEOUS_CORES))
-			return x86_sched_itmt_flags();
-	};
-
 	return 0;
 }
 
-- 
2.47.1

